### PR TITLE
feat: log which value did not match by reference (fixes #8)

### LINKED ISSUE #8: Log which property did not match by reference
Suppose I get a log message such as this:

screen shot 2017-11-19 at 13 24 05

It tells me that `EventsView` did not need to reload because current props and next props deep equal. However, as I am using `PureComponent`, I am not expecting the re-render to happen in the first place. It looks like some variables are being recreated.

The log should therefore include a list of properties that did not match by reference.

### PR DESCRIPTION
**Pull request by gajus**

### CODE REVIEW

**Comment by maicki** [comment]
@gajus Hey, sorry for the last response. I really appreciate your Issue and PR, I just didn't have time yet to review. I hope I can take a look over the next couple of days though. Thanks for being patient.

**Comment by mkraft** [comment]
This is a great call @gajus 👍

**Comment by maicki** [comment]
@gajus Thanks for working on it!

File: lib/defaultNotifier.js [hunk 1]
```diff
@@ -36,11 +36,17 @@ var notifyDiff = function notifyDiff(_ref) {
       console.warn(name + ': Value did not change. Avoidable re-render!');
       console.log('Before:', prev);
       console.log('After:', next);
+
+      for (const [name, value] of Object.entries(prev)) {
+        if (value !== next[name]) {
+          console.log('"' + name + '" property is not equal by reference');
+        }
+      }
```
Comment (OliverJAsh) [inline lib/defaultNotifier.js]: @maicki @gajus A few things:

- This appears to have only been added to the `lib` version of `defaultNotifier.js`, not `src/defaultNotifier.js`, so it won't be released :-(
- Should the comparison logic be captured by `classifyDiff`, as that's where all other comparison logic is defined?
- Can we add tests for this?

**Pull request merged**